File: cinema-backend/src/main/java/org/example/cinemabackend/cinema/application/dto/response/SeatResponse.java
package org.example.cinemabackend.cinema.application.dto.response;

import lombok.Builder;
import org.example.cinemabackend.cinema.core.domain.SeatStatus;
import org.example.cinemabackend.cinema.core.domain.SeatZone;

import java.util.List;

@Builder
public record SeatResponse(
        Long id,
        Integer seatRow,
        Integer seatColumn,
        SeatZone seatZone,
        List<SeatStatus> status
) {
}

File: cinema-customer-frontend/src/app/repertory/dtos/response/seat.response.ts
import {SeatZone} from "../../components/buy-ticket/enums/seat-zone";
import {SeatStatus} from "./seat-status";

export class SeatResponse {
  constructor(
    readonly id: number,
    readonly seatRow: number,
    readonly seatColumn: number,
    readonly seatZone: SeatZone,
    readonly status: SeatStatus[],
  ) {
  }
}

